docs/sagas.md

```diff
@@ -0,0 +1,239 @@
+      await store.delete('users', {
+        id: event.aggregateId
+      })
+    }
+  },
+  sideEffects: {
+    sendEmail: async (mail, content) => {
+      ...
+    }
+  }
+}
+```
+
+<!-- prettier-ignore-end -->
+
+This saga requires a new user to confirm their email address. If a user does not confirm the address in a week, the saga cancels the registration.
+
+## Define a Saga
+
+### Add a Saga to the Application
+
+You can define a saga in one of the following two ways:
+
+- In one source file as an object that contains the `handlers` and `sideEffects` objects.
+
+  ##### common/sagas/user-confirmation.saga.js:
+
+  ```js
+  export default {
+    handlers: {
+      // Event handlers implementation
+    }
+    sideEffects: {
+      // Side effects implementation
+    }
+  }
+  ```
+
+- In two separate files.
+
+  ##### common/sagas/user-confirmation.handlers.js:
+
+  ```js
+  export default {
+    // Event handlers implementation
+  }
+  ```
+
+  ##### common/sagas/user-confirmation.side-effects.js:
+
+  ```js
+  export default {
+    // Side effects implementation
+  }
+  ```
+
+### Initialize the Storage
+
+Every saga should define an `Init` function that initializes the saga's persistent storage:
+
+[mdis]: # '../tests/saga-sample/saga.js#init'
+
+```js
+Init: async ({ store }) => {
+  await store.defineTable('users', {
+    indexes: { id: 'string' },
+    fields: ['mail']
+  })
+},
+```
+
+### Handle Events
+
+An event handler function runs for every occurrence of a specific event. It has the following general structure:
+
+```js
+EVENT_NAME: async (
+  { executeCommand, scheduleCommand, store, sideEffects },
+  event
+) => {
+  // Event handler logic
+}
+```
+
+As a first argument, an event handler receives an object that provides access to the saga-related API. This API includes the following functions:
+
+- `executeCommand` - Sends a command with the specified payload to an aggregate.
+- `scheduleCommand` - Similar to `executeCommand` but delays the command execution until the specified moment in time.
+- `store` - Provides access to the saga's persistent store (similar to the Read Model store).
+- `sideEffects` - Provides access to the saga's side effect functions.
+
+### Send Aggregate Commands
+
+Use the `executeCommand` function to send aggregate commands as shown below:
+
+```js
+await executeCommand({
+  type: 'rejectUser',
+  aggregateName: 'User',
+  aggregateId,
+  payload: {
+    reason: 'user with same name already registered and confirmed'
+  }
+})
+```
+
+### Schedule Aggregate Commands
+
+The code sample below demonstrates how the command execution on a moment in time.
+
+```js
+await scheduleCommand(timestamp + 3600000, {
+  type: 'rejectUser',
+  aggregateName: 'User',
+  aggregateId,
+  payload: {
+    reason: 'user registration was not confirmed within allowed period'
+  }
+})
+```
+
+### Use Side Effects
+
+You should define all functions that have side effects in the `sideEffects` object.
```

Comment:
Change "You should define all functions that have" to "Define all functions that have"